src/cbc_sdk/platform/admin_grants.py

```diff
@@ -0,0 +1,446 @@
+            data = {'orgs': self._orgs, 'roles': self._roles, 'conditions': self._conditions}
+            profile = Grant.Profile(self._grant._cb, self._grant, None, data)
+            self._grant._profiles.append(profile)
+            self._grant.touch()
+            return profile
+
+    @classmethod
+    def _query_implementation(cls, cb, **kwargs):
+        """
+        Returns the appropriate query object for Grants.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            **kwargs (dict): Not used, retained for compatibility.
+
+        Returns:
+            GrantQuery: The query object for this alert type.
+        """
+        return GrantQuery(cls, cb)
+
+    def _refresh(self):
+        """
+        Rereads the grant data from the server.
+
+        Returns:
+            bool: True if refresh was successful, False if not.
+        """
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id)
+        resp = self._cb.get_object(url)
+        self._info = resp
+        self._profiles = [Grant.Profile(self._cb, self, prof.get('profile_uuid', None), prof)
+                          for prof in resp.get('profiles', [])]
+        self._last_refresh_time = time.time()
+        return True
+
+    def _update_object(self):
+        """
+        Updates the grant data on the server.
+
+        Returns:
+            str: The principal URN of this grant object.
+        """
+        save_info = copy.deepcopy(self._info)
+        if 'principal' not in save_info:
+            raise ApiError("principal must be specified for a Grant object")
+        save_info['profiles'] = [copy.deepcopy(prof._info) for prof in self._profiles if 'profile_uuid' in prof._info]
+
+        if all([key in save_info for key in ['created_by', 'updated_by', 'create_time', 'update_time']]):
+            log.debug("Updating the Grant object for principal {0:s}".format(save_info['principal']))
+            url = self.urlobject_single.format(self._cb.credentials.org_key, save_info['principal'])
+            ret = self._cb.put_object(url, save_info)
+        else:
+            log.debug("Creating a new Grant object for principal {0:s}".format(save_info['principal']))
+            url = self.urlobject.format(self._cb.credentials.org_key)
+            ret = self._cb.post_object(url, save_info)
+        update_return = self._refresh_if_needed(ret)
+
+        # add all profiles not yet part of grant
+        for profile in self._profiles:
+            if 'profile_uuid' not in profile._info:
+                profile._update_object()
+        return update_return
+
+    def _delete_object(self):
+        """Deletes the grant from the principal."""
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id)
+        ret = self._cb.delete_object(url)
+        self._refresh_if_needed(ret)
+
+    @classmethod
+    def create(cls, cb, principal):
```

Comment:
There should be an optional object here to provide the full grant object similar to how profiles is being done.

Reply:
Adding.